airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -18,8 +18,10 @@
 """Objects relating to sourcing connections & variables from Hashicorp Vault"""
 from typing import Optional
 
+from airflow.models.connection import Connection
 from airflow.providers.hashicorp._internal_client.vault_client import _VaultClient  # noqa
 from airflow.secrets import BaseSecretsBackend
+from airflow.secrets.local_filesystem import _create_connection
```

Comment:
We should extract the common part to the new package. This is problematic now, so I started discussions on the mailing list. See: https://lists.apache.org/thread.html/r713f180120d0a39b53567812eb5db34f992ec81979818d2175598b71%40%3Cdev.airflow.apache.org%3E

Reply:
In this particular case I can see this function existing in `airflow.models.connections` as `from_dict() -> Connection`. Transforming a dict to the object is the sort of function that usually exists with the class. I agree with the general case that something needs to be done about common functionality between providers.

Reply:
You can't add a new feature to the core because we want to be backwards compatible with Airflow 2.0.0.  I opened the discussions to loosen this file, because they do not conform to reality. I think we should maintain backward compatibility with the MINOR release, not the MAJOR. 

I invite you to the discussion on the mailing list if you would like to share your thoughts.

Reply:
It's an additional function on the Connection class so that retains backwards compatibility no?

Reply:
Secret Backend won't work with Airflow 2.0.0, so this is a breaking change to providers packages. This doesn't make some breaking changes to Airflow 2.0.0, but to providers packages it does.

Reply:
So you are saying that if this vault provider expects a `from_dict` method on `Connection` (added in 2.1) the provider won't work with 2.0, and but we've promised that it will.

should we perhaps then just merge this as is?  or perhaps duplicate this private function into this provider into a vault provider utils module with a deprecation warning?

Reply:
Yes. Exactly.

We should discuss the approach on the mailing list and choose the best solution that everyone accepts.  I personally think that we should mark this package as only supported by Airflow 2.1, because trying to maintain backward compatibility will limit our development possibilities. It should be normal for users that new packages/library versions may require a core version if we add new features.

Reply:
I started discussions on this subject on the mailing list, but I haven't checked it recently. I encourage you to discuss it yourself because then we will be able to work out a solution to this problem faster.

https://lists.apache.org/thread.html/r713f180120d0a39b53567812eb5db34f992ec81979818d2175598b71%40%3Cdev.airflow.apache.org%3E

Reply:
I have the impression that we have managed to reach a consensus. This change will be in the next provider package after Airflow 2.1 is released. There is a slight problem with the provider packages releases process. We always release providers packages from the main branch, so we cannot merge this PR until we release Airflow 2.1 and we will start releasing providers packages that will be compatible with this version.

In the meantime, we can create a new public method that will create connections from the dictionary. This method will be released in Airflow 2.1. 
https://github.com/apache/airflow/pull/15425